### PR TITLE
Migrate to JUnit 5

### PR DESCRIPTION
**Pull request by valfirst**

### CODE REVIEW

File: src/test/java/com/saucelabs/saucerest/SauceRESTTest.java [hunk 1]
```diff
@@ -572,23 +575,23 @@ public void testGetFullJobs() throws Exception {
...
             this.urlConnection.getRealURL().getPath()
         );
-        assertEquals(null, this.urlConnection.getRealURL().getQuery());
+        assertNull(this.urlConnection.getRealURL().getQuery());
 
     }
 
     @Test
-    public void testGetJobsLimit() throws Exception {
+    void testGetJobsLimit() {
```
Comment (sceiler) [inline src/test/java/com/saucelabs/saucerest/SauceRESTTest.java:592]: Why are the access modifiers for this and the following tests removed?

File: pom.xml [hunk 2]
```diff
@@ -56,6 +55,17 @@
             <url>https://oss.sonatype.org/service/local/staging/deploy/maven2/</url>
         </repository>
     </distributionManagement>
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.junit</groupId>
+                <artifactId>junit-bom</artifactId>
+                <version>5.7.0</version>
```
Comment (sceiler) [inline pom.xml:63]: I think this should be extracted into a property under properties.

File: pom.xml [hunk 3]
```diff
@@ -68,9 +78,14 @@
...
-            <artifactId>junit</artifactId>
-            <version>${junit.version}</version>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest</artifactId>
+            <version>2.2</version>
```
Comment (sceiler) [inline pom.xml:88]: Same as above

File: pom.xml [hunk 4]
```diff
@@ -223,6 +238,10 @@
                     <target>${java.version}</target>
                 </configuration>
             </plugin>
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>2.22.2</version>
```
Comment (sceiler) [inline pom.xml:243]: Same as above

**Review by sceiler — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 3 above)
Comment (sceiler) [reply pom.xml:88]: On hindsight nevermind.

File: pom.xml (hunk 4 above)
Comment (sceiler) [reply pom.xml:243]: Nevermind.

File: src/test/java/com/saucelabs/saucerest/SauceRESTTest.java (hunk 1 above)
Comment (valfirst) [reply src/test/java/com/saucelabs/saucerest/SauceRESTTest.java:592]: JUnit5 doesn't require `public` access modifier anymore and default access modifier improves readability:
https://rules.sonarsource.com/java/tag/junit/RSPEC-5786

Comment (sceiler) [reply src/test/java/com/saucelabs/saucerest/SauceRESTTest.java:592]: Thx, good to know.

**Review by sceiler — APPROVED** [review]

**Pull request merged**